src/alarms/notifications/NewNotification.js

```diff
@@ -340,94 +272,57 @@ class NewNotification extends Component {
+      )
+        .then(response => response.json())
+        .then(data => {
+          return data.results;
+        });
+      return assets;
+    } catch (e) {
+      throw new Error(e);
     }
   }
-  handleResetTimeseriesAsset(timeseriesAsset) {
-    this.handleSetTimeseriesAsset("");
-  }
-  handleSetTimeseriesNestedAsset(timeseriesNestedAsset) {
-    // fetch nested asset if any by following asset and number
-    // filters as nested assets for groundwaterstations
-    // add all nested assets!
-    // this.setState({ timeseriesNestedAsset: "" });
-    // this.setState({ timeseriesNestedAssets: [] });
-    fetchNestedAssets(
-      this.state.timeseriesAssetType,
-      this.state.timeseriesAssetId
-    ).then(data => {
-      console.log(
-        "NewNotification handleSetTimeseriesNestedAsset nestedAsset data",
-        data
-      );
-      let nestedAssets = [];
-      for (var i = 0; i < data.length; i++) {
-        if (data[i].code) {
-          nestedAssets.push(data[i].code);
-        }
-      }
-      // choices of SelectBoxSearch for timeserie nested assets
-      this.setState({ timeseriesNestedAssets: nestedAssets });
-      console.log(
-        "NewNotification handleSetTimeseriesNestedAsset nestedAssets",
-        nestedAssets
-      );
-      // choice of SelectBoxSearch for timeserie nested asset
-      this.setState({ timeseriesNestedAsset: timeseriesNestedAsset });
+  async handleSetTimeseriesAsset(assetObj) {
+    this.setState({
+      selectedTimeseriesAssetFromSearchEndpoint: assetObj
     });
-    // check for all nested assets in client
-    // Also set timeseries uuid
-  }
-  validateTimeseriesNestedAsset(str) {
-    if (str && str.length > 1) {
-      return true;
-    } else {
-      return false;
+    try {
+      // Set page_size to 100000, same as in Raster.js
+      const asset = await fetch(
+        `/api/v3/${assetObj.entity_name}s/${assetObj.entity_id}/?page_size=100000`,
+        {
+          credentials: "same-origin"
+        }
+      )
+        .then(response => response.json())
+        .then(data => {
+          return data;
+        });
+      this.setState({
+        selectedTimeseriesAssetFromAssetEndpoint: asset
+      });
+    } catch (e) {
+      throw new Error(e);
     }
   }
-  handleResetTimeseriesNestedAsset(timeseriesNestedAsset) {
-    this.handleSetTimeseriesNestedAsset("");
-  }
-  handleSetTimeseriesUuid(timeseriesUuid) {
-    this.setState({ timeseriesUuid: timeseriesUuid });
+  validateTimeseriesAsset(obj) {
+    return obj.title && obj.entity_id && obj.entity_name;
   }
-  validateTimeseriesUuid(str) {
-    if (str && str.length > 1) {
-      return true;
-    } else {
-      return false;
-    }
-  }
-  handleResetTimeseriesUuid(timeseriesUuid) {
-    this.handleSetTimeseriesUuid("");
+  handleResetTimeseriesAsset() {
+    this.setState({
+      foundTimeseriesAssetsSearchEndpoint: [],
+      selectedTimeseriesAssetFromSearchEndpoint: {},
+      selectedTimeseriesAssetName: ""
```

Comment:
selectedTimeseriesAssetName kan denk ik weg, maar daarvoor in de plaats misschien "selectedTimeseriesAssetFromAssetEndpoint